### PR TITLE
Produce test coverage reports

### PR DESCRIPTION
**Pull request by jamiebrynes7**
#### Description

When running our tests, we will now output an `.xml` file with test coverage information. This will later be consumed by a service. I've also removed the Playmode tests for the `playground` project as we have none of those. I'm pretty sure that the existing Playmode tests can be ported to Editmode in the `test-project` too. This should cut down on runtime :) 

#### Tests

Observed artifacts in CI. 

#### Documentation

- [ ] Changelog

### CODE REVIEW

**Comment by jamiebrynes7** [comment]
> This should cut down on runtime.

Cuts approximately 2 minutes off the test step 🎉

File: ci/test.sh [hunk 1]
```diff
@@ -63,6 +60,9 @@ echo "--- Testing Unity: Test Project Playmode :joystick:"
 pushd "test-project"
     dotnet run -p "${PROJECT_DIR}/.shared-ci/tools/RunUnity/RunUnity.csproj" -- \
         -batchmode \
+        -enableCodeCoverage \
+        -coverageResultsPath "${COVERAGE_RESULTS_PATH}/test-project" \
```
Comment (zeroZshadow) [inline ci/test.sh:64]: Will it be oke that this and the editmode tests go to the same location?

Comment (jamiebrynes7) [reply ci/test.sh:64]: It creates an `Editmode/Playmode` folder for each 😄
